pkg/controller/tidbcluster/pod_control.go

```diff
@@ -213,6 +219,52 @@ func (c *PodController) getPDClient(tc *v1alpha1.TidbCluster) pdapi.PDClient {
 	return pdClient
 }
 
+func (c *PodController) syncPDPod(ctx context.Context, pod *corev1.Pod, tc *v1alpha1.TidbCluster) (reconcile.Result, error) {
+	key, value, ok := needEvictLeader(pod)
```

Comment:
Discussed with @luohao offline, should we use `EvictLeaderAnnKeys` to mark for restart, or we should use a new dedicated key for that purpose? Currently that `EvictLeaderAnnKeys` key indicates "transfer all TiKV raft store leaders on this TiKV instance to other instances", where for PD it means "transfer the PD leader to other nodes". They're slightly different semantically, but if we use that annotation for components like TiDB, it could be confusing semantically since there's no "leaders" anymore. Any comments or suggestion @csuzhangxc?

Reply:
I prefer to use different annotation keys. for PD, I think `transfer leader` is better than `evict leader`.

Reply:
Thanks @hanlins and @csuzhangxc !

This PR is ready for review. PTAL!